File: mali/Mali.kt
fun Mali(lines : MutableList<String>) : String {
    var min_a = -1
    var min_b = -1
    var max_a = 0
    var max_b = 0

    lines.removeAt(0)

    var retString = ""

    lines.forEach { line ->
        val nums = line.split(" ").map { it.toInt() }

        if (min_a == -1 || min_a > nums[0]) {
            min_a = nums[0]
        }

        if (min_b == -1 || min_b > nums[1]) {
            min_b = nums[1]
        }

        if (max_a < nums[0]) {
            max_a = nums[0]
        }

        if (max_b < nums[1]) {
            max_b = nums[1]
        }


        val pot1 = min_a + max_b
        val pot2 = max_a + min_b

        if (pot1 > pot2) {
            retString += pot1.toString() + "\n"
        } else {
            retString += pot2.toString() + "\n"
        }

    }
    return retString.trim()
}


fun main() {
    val lines = generateSequence(::readLine).toMutableList()
    println(Mali(lines))
}

File: mali/mali.cpp
using namespace std;
#include <string>
#include <sstream>
#include <vector>
#include <iostream>



int main() {
    int numCases;
    scanf("%d", &numCases);

    int min_a = -1;
    int min_b = -1;
    int max_a = 0;
    int max_b = 0;


    for (int i = 0; i < numCases; i++) {
        int num1, num2;

        scanf("%d", &num1);
        scanf("%d", &num2);

        min_a = (min_a == -1 || num1 < min_a) ? num1 : min_a;
        min_b = (min_b == -1 || num2 < min_b) ? num2 : min_b;

        max_a = (num1 > max_a) ? num1 : max_a;
        max_b = (num2 > max_b) ? num2 : max_b;

        int pot1 = max_a + min_b;
        int pot2 = min_a + max_b;

        int biggest = (pot1 > pot2) ? pot1 : pot2;

        cout << biggest << "\n";

    }
}
